### PR TITLE
Closes #338 | Created DataLoader for IndonesianNMT

### LINKED ISSUE #338: Create dataset loader for IndonesianNMT
Dataloader name: `indonesiannmt/indonesiannmt.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?indonesiannmt

|     Dataset| indonesiannmt |
|-------------|---|
| Description | Repository containing datasets for data automatically generated from gpt-3.5-turbo and text-davinci-003 used in the work "Replicable Benchmarking of Neural Machine Translation (NMT) on Low-Resource Local Languages in Indonesia" |
| Subsets | - |
| Languages | ind, jav, min, sun, ban |
| Tasks | Machine Translation |
| License | Unknown (unknown) |
| Homepage | https://github.com/luckysusanto/IndonesianNMT |
| HF URL | - |
| Paper URL | https://arxiv.org/abs/2311.00998 |

### PR DESCRIPTION
**Pull request by luckysusanto**
Closes #338 

### Checkbox
- [X] Confirm that this PR is linked to the dataset issue.
- [X] Create the dataloader script `seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py` (please use only lowercase and underscore for dataset naming).
- [X] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [X] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [X] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [X] Confirm dataloader script works with `datasets.load_dataset` function.
- [X] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by luckysusanto** [comment]
These data were used to train XLM for machine translation.
As there are 4 parallel corpora and 2 monolingual corpora, a total of 2 (default schema, goes to parallel_ind_jav) + 12 Builder Configs are set.

The other 12 Custom Builder Configs are:
```indonesiannmt-{modifier}_{schema}```
where
```modifier = {'mono_ind', 'mono_jav', 'parallel_ind_jav', 'parallel_ind_min', 'parallel_ind_sun', 'parallel_ind_ban'}
schema = {'source', 'seacrowd_t2t'}```

**Comment by MJonibek** [comment]
@luckysusanto I think it is better to redo configurations for dataloader. I don`t think it is good idea to add monolingual files for machine translation tasks. It is better just to remove them (or maybe add them under another task, @holylovenia what do you think). 

Also, it is better to rename modifiers from parallel_{lang1}_{lang2} to {lang1}_{lang2} by deleting `parallel` prefixes.

**Review by holylovenia — CHANGES_REQUESTED** [review]
I agree with @MJonibek's suggestion. It's better to have the monolingual data removed or separate from the MT subsets. I saw from [the monitor sheet](https://docs.google.com/spreadsheets/d/1ibbywsC1tQ_sLPX8bUAjC-vrTrUqZgZA46W_sxWw4Ss/edit?usp=sharing) that you are the dataset owner, @luckysusanto. Could you please let us know what task this monolingual data is created for? Is it `Tasks.LANGUAGE_MODELING`?

The MT `seacrowd` subsets should follow this convention: `{datasetname}_{lang1}_{lang2}_seacrowd_t2t`, e.g., `indonesiannmt`.

**Comment by luckysusanto** [comment]
Noted, I will change the schemes of the dataloader @MJonibek 

@holylovenia the monolingual data was used for unsupervised machine learning on the XLM architecture. That was why I still left it there under the machine learning task, even though it is unintuitive.

I'll check on what the ssp schema is and will implement the dataloader in the near future if it works!
Thanks for the inputs! ^^

**Comment by holylovenia** [comment]
> luckysusanto

You can take a look at [the `ssp` schema](https://github.com/SEACrowd/seacrowd-datahub/blob/master/seacrowd/utils/schemas/self_supervised_pretraining.py), used for `Tasks.LANGUAGE_MODELING` aka unlabelled data.

Sure sure, take your time and let us know if you would like to discuss. Nice to e-meet you after AACL! 😄

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (holylovenia) [inline seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:73]: `"-"` --> `"_"`

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (holylovenia) [inline seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:75]: `"-"` --> `"_"`

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (holylovenia) [inline seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:185]: Let's use the 3-letter ISO code of the language, e.g., `"ind"` instead of `"Indonesian"`.

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (holylovenia) [inline seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:186]: Let's use the 3-letter ISO code of the language, e.g., `"jav"` instead of `"Javanese"`.

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (holylovenia) [inline seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:89]: There's no need for `indonesiannmt_source` if it only loads the Javanese data. We can remove this subset.

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (holylovenia) [inline seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:96]: There's no need for `indonesiannmt_seacrowd_t2t` if it only loads the Javanese data. We can remove this subset.

**Review by holylovenia — COMMENTED** [review]
Hi @luckysusanto, the dataloader works well. I have some questions/suggestions to improve it:

1. Is there any reason why you decide not to include the monolingual data in the dataloader? Previously, we have discussed that it might be included using the `ssp` schema.

If you decide to include it, then we can add `indonesiannmt_mono_{lang}_source` and `indonesiannmt_{lang}_seacrowd_ssp` and `Tasks.SELF_SUPERVISED_PRETRAINING`.

**Comment by MJonibek** [comment]
> 1. Is there any reason why you decide not to include the monolingual data in the dataloader? Previously, we have discussed that it might be included using the `ssp` schema. (cc: What do you think, @MJonibek?)

I also think it would be great to add `ssp` schema for monolingual data sets

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (luckysusanto) [reply seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:89]: I recall failing the test case if I didn't create the "indonesiannmt_source" schema though. I'll try to remove it and see what happens though!

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (luckysusanto) [reply seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:186]: Gotcha

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (luckysusanto) [reply seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:75]: Aight

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (luckysusanto) [reply seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:73]: Will do

**Comment by luckysusanto** [comment]
`Is there any reason why you decide not to include the monolingual data in the dataloader? Previously, we have discussed that it might be included using the ssp schema. (cc: What do you think, @MJonibek?)`

My bad! Yeah, it skipped my mind, entirely my fault. 
I'll implement it while also fixing the styling issues you have mentioned @holylovenia.

Will ping you guys later when I fix it, ETA: Wednesday

File: seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py
Comment (holylovenia) [reply seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:89]: You can either declare a default schema name or use a subset id when running the unit test, e.g., `python -m tests.test_seacrowd seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py --subset_id=indonesiannmt_ind_jav`.

Comment (luckysusanto) [reply seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py:89]: ah, my bad, thanks for the correction

**Comment by luckysusanto** [comment]
Requesting re-review:
@holylovenia @MJonibek 

Update Log:
SSP task implemented for monolingual data
Changes requested by holy was implemented!

Test ran successfully:
python -m tests.test_seacrowd seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py --schema=T2T --subset_id=indonesiannmt_ind_jav
python -m tests.test_seacrowd seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py --schema=SSP --subset_id=indonesiannmt_ind

**Review by MJonibek — APPROVED** [review]
@luckysusanto LGTM, great work and thank you. The only problem is formatting. Can you run 

`make check_file=seacrowd/sea_datasets/indonesiannmt/indonesiannmt.py `

If you still have problems with running this command, I can do it and commit before merging.

**Review by holylovenia — APPROVED** [review]
LGTM!!! Thanks a lot, @luckysusanto. Will merge once you address @MJonibek's request.

**Comment by MJonibek** [comment]
@holylovenia, I`ve done formatting by myself, can merge

**Pull request merged**
